trendmicro_apex/icon_trendmicro_apex/actions/add_to_udso_list/action.py

```diff
@@ -0,0 +1,93 @@
+import komand
+from .schema import AddToUdsoListInput, AddToUdsoListOutput, Input, Output, Component
+from komand.exceptions import PluginException
+from requests.exceptions import RequestException
+# Custom imports below
+from icon_trendmicro_apex.util.util import get_expiration_utc_date_string
+import ipaddress
+import json
+import requests
+import re
+
+
+class AddToUdsoList(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='add_to_udso_list',
+                description=Component.DESCRIPTION,
+                input=AddToUdsoListInput(),
+                output=AddToUdsoListOutput())
+        self.api_path = '/WebApp/api/SuspiciousObjects/UserDefinedSO'
+        self.api_http_method = 'PUT'
+
+
+    def run(self, params={}):
+        payload_notes = ''
+        user_notes = params.get(Input.NOTES)
+        if user_notes:
+            if len(user_notes) > 256:
+                self.logger.warning("Notes passed in too long, truncated to 256 characters")
+            payload_notes = user_notes[:256]
+        payload_scan_action = params.get(Input.SCAN_ACTION)
+        payload_content = params.get(Input.CONTENT).lower()
+        payload_type = params.get(Input.DATA_TYPE)
+        num_days = params.get(Input.EXPIRY_DATE, 30)
+        payload_expiry_date = get_expiration_utc_date_string(int(num_days))
+
+        if payload_type == 'FILE_SHA1':
+            if len(payload_content) > 40:
```

Comment:
I'd consider making the `40` a constant, like SHA_LENGTH_MAX or something so it is easier to reuse.